SignalMessaging/utils/ProximityMonitoringManager.swift

```diff
@@ -39,20 +49,44 @@ public class OWSProximityMonitoringManagerImpl: NSObject, OWSProximityMonitoring
         }
     }
 
+    @objc
+    public func setup() {
+        NotificationCenter.default.addObserver(self, selector: #selector(proximitySensorStateDidChange(notification:)), name: .UIDeviceProximityStateDidChange, object: nil)
+    }
+
+    @objc
+    func proximitySensorStateDidChange(notification: Notification) {
+        Logger.debug("")
+        // This is crazy, but if we disable `device.isProximityMonitoringEnabled` while
+        // `device.proximityState` is true (while the device is held to the ear)
+        // then `device.proximityState` remains true, even after we bring the phone
+        // away from the ear and re-enable monitoring.
+        //
+        // To resolve this, we wait to disable proximity monitoring until `proximityState`
+        // is false.
+        if self.device.proximityState {
+            self.add(lifetime: self)
+        } else {
+            self.remove(lifetime: self)
```

Comment:
`proximitySensorStateDidChange(...)` is presumably called on the main thread.  `add()`/`remove()`/`reconcile()` do their work on the serial queue.  I think there's a (rare, possibly inconsequential) race wherein `reconcile()` is called _after_ `proximityState` becomes true but before the `add(lifetime:self)` is done.  

I'm going to modify this PR to use `NSLock` or `objc_sync_enter()` to avoid this race.